### PR TITLE
Wire up calculate-min-fee and calculate-min-value

### PR DESCRIPTION
**Pull request by Jimbo4350**

### CODE REVIEW

File: cardano-api/src/Cardano/Api/ProtocolParameters.hs [hunk 1]
```diff
@@ -1323,3 +1331,36 @@ fromAlonzoPParams
...
+-- mininimum UTxO value derived from the 'ProtocolParameters'
+calcMinimumDeposit :: ProtocolParameters -> Value -> Lovelace
+calcMinimumDeposit pp v  =
+  case protocolParamMinUTxOValue pp of
+    Nothing ->
+      case protocolParamUTxOCostPerWord pp of
+        Nothing ->
+          error "calcMinimumDeposit: uTxOCostPerWord and minUTxOValue protocol parameters were not found"
+        Just (Lovelace cPerWord) ->
+          -- TODO: Alonzo - ledger should expose a function to calculate the required deposit
```
Comment (newhoggy) [inline cardano-api/src/Cardano/Api/ProtocolParameters.hs:1344]: TODO item

File: cardano-api/src/Cardano/Api/ProtocolParameters.hs [hunk 2]
```diff
@@ -1323,3 +1331,36 @@ fromAlonzoPParams
...
+          Lovelace $ max adaOnlyUTxOSize (utxoEntrySizeWithoutVal + Val.size (toMaryValue v)) * cPerWord
+    Just minUTxOValue ->
+      fromShelleyLovelace $ Shelley.scaledMinDeposit (toMaryValue v) (toShelleyLovelace minUTxOValue)
+
+ where
+    -- TODO: Copied from ledger. Ledger should expose a function for this
```
Comment (newhoggy) [inline cardano-api/src/Cardano/Api/ProtocolParameters.hs:1350]: TODO item for discussion

Comment (JaredCorduan) [reply cardano-api/src/Cardano/Api/ProtocolParameters.hs:1350]: this is a part of `utxoEntrySize` mentioned above

File: cardano-api/src/Cardano/Api/ProtocolParameters.hs (hunk 1 above)
Comment (JaredCorduan) [reply cardano-api/src/Cardano/Api/ProtocolParameters.hs:1344]: The size is calculated here:

https://github.com/input-output-hk/cardano-ledger-specs/blob/ecb7275b2de624cede629fec2c4e9237639425e3/alonzo/impl/src/Cardano/Ledger/Alonzo/Rules/Utxo.hs#L104

and is multiplied by `adaPerUTxOWord` here:

https://github.com/input-output-hk/cardano-ledger-specs/blob/ecb7275b2de624cede629fec2c4e9237639425e3/alonzo/impl/src/Cardano/Ledger/Alonzo/Rules/Utxo.hs#L394

Comment (Jimbo4350) [reply cardano-api/src/Cardano/Api/ProtocolParameters.hs:1344]: @JaredCorduan That calculates the size of the tx and then the cost (via multiplication by `adaPerUTxOWord`).  We want to only calculate the minimum deposit for a multi asset value.

Comment (JaredCorduan) [reply cardano-api/src/Cardano/Api/ProtocolParameters.hs:1344]: just let me know, we could certainly add something for you. did you want a function that takes a value and a maybe hash? or are you good now?

**Comment by Jimbo4350** [comment]
Subsumed by #2612

**Pull request closed without merging**